src/System.Console/src/System/IO/SyncTextReader.cs

```diff
@@ -11,7 +11,6 @@ namespace System.IO
     internal sealed class SyncTextReader : TextReader
     {
         private readonly TextReader _in;
-        private readonly object _methodLock = new object();
```

Comment:
We should definitely add a comment to the implementation.

We should also add a test, e.g.
```C#
[Fact]
public void SyncTextWriterLockedOnThis()
{
    TextWriter oldWriter = Console.Out;
    try
    {
        var newWriter = new CallbackTextWriter();
        Console.SetOut(newWriter);
        TextWriter syncWriter = Console.Out;

        bool called = false;
        newWriter.Callback = _ =>
        {
            Assert.True(Monitor.IsEntered(syncWriter));
            called = true;
        };
        Console.Write("c");
        Assert.True(called);
    }
    finally
    {
        Console.SetOut(oldWriter);
    }
}

private sealed class CallbackTextWriter : TextWriter
{
    internal Action<char> Callback;

    public override Encoding Encoding { get { return Encoding.UTF8; } }

    public override void Write(char value) { Callback(value); }
}
```